engine/src/org/pentaho/di/trans/steps/rest/Rest.java

```diff
@@ -302,7 +320,8 @@ public boolean processRow( StepMetaInterface smi, StepDataInterface sdi ) throws
 
     Object[] r = getRow(); // Get row from input rowset & set row busy!
 
-    if ( r == null ) {
+    logBasic(Arrays.toString(r));
+    if ( r == null && !first ) {
```

Comment:
Not sure this should be at the Basic level, nor should it be logged if the row is null. If we want to log that there were no rows, we should do that at the Detailed or Debug level.